DataObjects/DataObjects/GeantEventEntry.h

```diff
@@ -0,0 +1,23 @@
+#ifndef DATAOBJECTS_GEANTEVENTENTRY_H
+#define DATAOBJECTS_GEANTEVENTENTRY_H
+
+//Include files
+#include "GaudiKernel/DataObject.h"
+#include "G4Event.hh"
+
+class GAUDI_API GeantEventEntry : public DataObject {
+ public:
+  GeantEventEntry() : m_event(0){};
+   virtual ~GeantEventEntry() {if (m_event!=0) delete m_event;};
```

Comment:
this should read:

```c++
virtual ~GeantEventEntry() {
delete m_event;
m_event = nullptr;
};
```
otherwise, double-deletes will happen.
also, probably one should use a smart pointer of some kind very early on in FCC-SW.